Data Structures/Disjoint Set Union/disjoint_set_union.cpp

```diff
@@ -0,0 +1,56 @@
+#include<iostream>
+using namespace std;
+//TIME COMPLEXITY: As the loop iterates through N elements once so it takes O(N) time.
+void initialize(int A[] ,int n){ // initialize the array
+    for(int i=0;i<n;i++)
+        A[i] = i;
+}
+
+// returns true if x and y are connected, else returns false
+//TIME COMPLEXITY: O(1) time needed as only a comparison need to be done
```

Comment:
Implement find as returning the representative element of the disjoint set rather than true or false.

Reply:
@dhairyabaweja Let us know if you are still working. Otherwise, this PR should be closed.